staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -1086,6 +1086,24 @@ func TestPriorityAndFairnessWithPanicRecoveryAndTimeoutFilter(t *testing.T) {
 	})
 }
 
+func TestDisabledApfWatermarkMainteneace(t *testing.T) {
+	server := newApfServerWithSingleRequest(t, decisionQueuingExecute, false)
+	defer server.Close()
+
+	// when we disable watermarks that mean that we do not write to waitingMark variable
+	// so we can set it to nil and expect to panics
+	waitingMark = nil
+	defer func() {
```

Comment:
I don't think this panic handler is working as intended. If I call newApfServerWithSingleRequest with true instead of false (resulting in a panic), it doesn't recover.